presto-main/src/test/java/io/prestosql/execution/TestNodeScheduler.java

```diff
@@ -111,10 +124,24 @@ public void tearDown()
         finalizerService.destroy();
     }
 
+    // Test exception throw when no nodes available to schedule
+    @Test
+    public void testAssignmentWhenNoNodes()
+    {
+        Set<Split> splits = new HashSet<>();
+        splits.add(new Split(CONNECTOR_ID, new TestSplitRemote(), Lifespan.taskWide()));
+
+        assertThatThrownBy(() -> nodeSelector.computeAssignments(splits, ImmutableList.copyOf(taskMap.values())))
```

Comment:
BTW @electrum just added `PrestoExceptionAssert.assertPrestoExceptionThrownBy` which makes this even easier.